### PR TITLE
Catalog Ingestion DB Schema / Models and Configuration

### PR DESCRIPTION
**Pull request by garmashdev**
# Description
Implemented database schema, base models/resources/repositories for CRUD operations. Added system configuration and config.xml with default values.

https://app.asana.com/0/1201931884901947/1201906341917644/f

#changelog Catalog Ingestion DB Schema / Models and Configuration

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 

- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my ticket link and provided a changelog message.

### CODE REVIEW

File: Api/Data/ProductEventInterface.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    public const PRODUCT_ID = 'product_id';
+
+    public const TYPE = 'type';
+
+    public const CREATED_AT = 'created_at';
+
+    /**
+     * Allowed product event operation types
+     */
+    public const TYPE_UNKNOWN = 'unknown';
```
Comment (vitaliyreznikov) [inline Api/Data/ProductEventInterface.php:39]: Why do we need unknown type?

**Review by vitaliyreznikov — APPROVED** [review]

**Comment by vitaliyreznikov** [comment]
@mikeshutov-bolt Can you please check wording in this PR?

**Review by vitaliyreznikov — APPROVED** [review]

File: Api/Data/ProductEventInterface.php (hunk 1 above)
Comment (vitaliyreznikov) [reply Api/Data/ProductEventInterface.php:39]: I think we are not going to use it, let's remove at least for now.

Comment (garmashdev) [reply Api/Data/ProductEventInterface.php]: Removed.

**Review by vitaliyreznikov — APPROVED** [review]
LGTM

**Pull request merged**